tutorverse-client/src/components/ChatHome/ChatHome.jsx

```diff
@@ -1,49 +1,110 @@
 import { useNavigate } from "react-router-dom";
+import axios from "axios";
 import { useEffect, useState } from "react";
 import TextField from "@mui/material/TextField";
 import ArrowBackIosIcon from "@mui/icons-material/ArrowBackIos";
 import Button from "@mui/material/Button";
+import socketIO from "socket.io-client";
 import "./ChatHome.css";
 
-const messageList = [];
-export default function ChatHome({ socket, studentId, tutorId }) {
-  const [message, setMessage] = useState("");
+export default function ChatHome({ user }) {
+  const [tutors, setTutors] = useState([]);
+  const [messages, setMessages] = useState([]);
+  const [selectedUser, setSelectedUser] = useState(null);
+  const [newMessage, setNewMessage] = useState("");
+  const [socket, setSocket] = useState(null);
   const navigate = useNavigate();
 
+  //GET request for tutorlist
+  useEffect(() => {
+    const fetchTutors = async () => {
+      const response = await axios.get("http://localhost:3000/tutors", {
+        withCredentials: true,
+      });
+      setTutors(response.data.list);
+      console.log(response.data.list);
+    };
+    fetchTutors();
+  }, []);
+
   const handleBackClick = () => {
     navigate("/dashboard");
   };
 
-  // Sending a message
-  const handleSend = () => {
-    if (message.trim() !== "") {
-      messageList.push({ user: 0, text: message });
-      console.log(messageList);
-      const privateRoom = `${studentId}-${tutorId}`; // Generate a unique room name
-      socket.emit("receiving message", {
-        content: message,
-        to: 16,
-        room: privateRoom,
+  useEffect(() => {
+    if (user) {
+      socketIO.auth = { userId: user.id };
+      console.log(user.id);
+
+      const newSocket = socketIO("http://localhost:4000", {
+        auth: { userId: user.id },
       });
-      setMessage("");
+      setSocket(newSocket);
     }
-  };
+  }, [user]);
 
-  const handleInputChange = (e) => {
-    setMessage(e.target.value);
-  };
-
-  // Receiving a message
   useEffect(() => {
     if (socket) {
-      console.log("Listening for private message");
-      socket.on("private message", ({ content, from }) => {
-        console.log(content, from);
-        messageList.push({ user: 1, text: content });
-        console.log(messageList);
+      //Listener for receiving messages
+      socket.on("privateMessage", (data) => {
+        setMessages((prevMessages) => [...prevMessages, data]);
       });
+
+      return () => {
+        socket.off("privateMessage");
+      };
     }
-  }, [socket]);
+  });
```

Comment:
You can probably combine these two effects into 1 effect. Also the dependency array is missing in the second one, which causes it to run every render.